File: app/src/main/java/com/ywwynm/everythingdone/utils/FontCache.java
package com.ywwynm.everythingdone.utils;

import android.content.Context;
import android.graphics.Typeface;

import java.util.HashMap;

/**
 * Created by ywwynm on 2016/3/16.
 * cache fonts to prevent memory leak
 */
public class FontCache {

    public static final String TAG = "FontCache";

    private FontCache() {}

    private static HashMap<String, Typeface> fontCache = new HashMap<>();

    public static Typeface get(String name, Context context) {
        Typeface tf = fontCache.get(name);
        if (tf == null) {
            tf = Typeface.createFromAsset(context.getAssets(), name);
            fontCache.put(name, tf);
        }
        return tf;
    }

}

File: app/src/main/java/com/ywwynm/everythingdone/utils/FontCache.kt
package com.ywwynm.everythingdone.utils

import android.content.Context
import android.graphics.Typeface

import java.util.HashMap

/**
 * Created by ywwynm on 2016/3/16.
 * Translated to Kotlin by ywwynm and Claude Opus 4.7 on 2026/5/20.
 * cache fonts to prevent memory leak
 */
object FontCache {

    const val TAG: String = "FontCache"

    private val fontCache: HashMap<String?, Typeface?> = HashMap()

    @JvmStatic
    fun get(name: String?, context: Context?): Typeface? {
        var tf: Typeface? = fontCache[name]
        if (tf == null) {
            tf = Typeface.createFromAsset(context!!.assets, name)
            fontCache[name] = tf
        }
        return tf
    }

}
